OutOfSchool/OutOfSchool.WebApi/Common/Resources/Describers/ImagesErrorDescriber.cs

```diff
@@ -0,0 +1,124 @@
+﻿using Microsoft.AspNetCore.Identity;
+
+namespace OutOfSchool.WebApi.Common.Resources.Describers
+{
+    public class ImagesErrorDescriber
```

Comment:
maybe use a factory pattern with a method that receives error type and returns operationerror object?

Reply:
Sounds really cool but in this way we won't be able to build messages dynamically (I mean calling methods with some parameters and then use string formatting/interpolation to create something new, I've used it in ExceedingCountOfImagesError(int countOfImages)). Of course in this case we could create a little bit more general message and delete parameters. So we should discuss it :)

Reply:
```
public enum OperationError : byte
{
		[AppResourcesKey("txtDefaultError")]
		DefaultError = 0,

		[AppResourcesKey("txtUploadingError")]
		UploadingError = 1,
}
```

```
public class AppResourcesKeyAttribute : Attribute
{
		#region Constructor

		/// <summary>
		/// Initializes a new instance of the <see cref="AppResourcesKeyAttribute"/> class.
		/// </summary>
		/// <param name="appResourcesKey">The application resources key.</param>
		public AppResourcesKeyAttribute(string appResourcesKey) => AppResourcesKey = appResourcesKey;

		#endregion

		#region Public properties

		/// <summary>
		/// Gets or sets the application resources key.
		/// </summary>
		public string AppResourcesKey { get; set; }

		#endregion
}
```
```
/// <summary>
		/// Returns text from AppResources by AppResourceKey
		/// </summary>
		/// <param name="en">target enum</param>
		/// <returns>AppResources by AppResourceKey</returns>
		public static string GetAppResource(this Enum en)
		{
			var appResourceKeyAttribute = en.GetAttribute<AppResourcesKeyAttribute>();

			return appResourceKeyAttribute == null
				? string.Empty
				: AppResources.ResourceManager.GetString(appResourceKeyAttribute.AppResourcesKey);
		}
```
we can do something like this and use the factory method))
so we will be able to add new errors easily))